html/pfappserver/lib/pfappserver/Form/Config/SelfService.pm

```diff
@@ -80,6 +80,18 @@ has_field 'device_registration_allowed_devices' =>
    fingerbank_model => "fingerbank::Model::Device",
   );
 
+has_field 'device_registration_list_roles' =>
```

Comment:
What happens when you set this and a device registration role:
https://github.com/inverse-inc/packetfence/blob/611a4ea93c3b2909273fdbe742f412a69c7c03a8/html/pfappserver/lib/pfappserver/Form/Config/SelfService.pm#L49

I think that instead of creating a new field, lets just expand the existing one and when you put more than 1 role it lets the user pick it. If you put just 1 role, it forces it on the node and 0 role, it uses the one of the user

Reply:
Yes good catch